tests/network/general/test_bridge_marker.py

```diff
@@ -127,11 +127,17 @@ def non_homogenous_bridges(worker_node1, worker_node2):
             yield bridgemarker2_ncp, bridgemarker3_ncp
 
 
-def _assert_failure_reason_is_bridge_missing(pod, bridge):
-    cond = pod.instance.status.conditions[0]
-    missing_resource = bridge.resource_name
-    assert cond.reason == "Unschedulable"
-    assert f"Insufficient {missing_resource}" in cond.message
+def _assert_failure_reason_is_bridge_missing(vmi, bridge):
+    requested_condition = "PodScheduled"
+    for cond in vmi.instance.status.conditions:
+        if cond.type == requested_condition:
+            assert cond.status == "False"
+            assert cond.reason == "Unschedulable"
+            assert f"Insufficient {bridge.resource_name}" in cond.message
```

Comment:
As a follow-up, can you please add conditions lookup function in the wrapper to look a condition based on optional status, reason and message?